ros_gz_sim/launch/gz_sim.launch.py.in

```diff
@@ -14,20 +14,102 @@
+            package_share_path = get_package_share_directory(package_name)
+            package_file_path = os.path.join(package_share_path, PACKAGE_MANIFEST_FILENAME)
+            if os.path.isfile(package_file_path):
+                try:
+                    package = parse_package(package_file_path)
+                except InvalidPackage:
+                    continue
+                for export in package.exports:
+                    if export.tagname == 'gazebo_ros':
+                        if 'gazebo_model_path' in export.attributes:
+                            xml_path = export.attributes['gazebo_model_path']
+                            xml_path = xml_path.replace('${prefix}', package_share_path)
+                            gazebo_model_path.append(xml_path)
+                        if 'plugin_path' in export.attributes:
+                            xml_path = export.attributes['plugin_path']
+                            xml_path = xml_path.replace('${prefix}', package_share_path)
+                            gazebo_plugin_path.append(xml_path)
+                        if 'gazebo_media_path' in export.attributes:
+                            xml_path = export.attributes['gazebo_media_path']
+                            xml_path = xml_path.replace('${prefix}', package_share_path)
+                            gazebo_media_path.append(xml_path)
+
+        gazebo_model_path = os.pathsep.join(gazebo_model_path)
+        gazebo_plugin_path = os.pathsep.join(gazebo_plugin_path)
+        gazebo_media_path = os.pathsep.join(gazebo_media_path)
+
+        return gazebo_model_path, gazebo_plugin_path, gazebo_media_path
+
+
 def launch_gz(context, *args, **kwargs):
-    env = {'GZ_SIM_SYSTEM_PLUGIN_PATH':
-           ':'.join([environ.get('GZ_SIM_SYSTEM_PLUGIN_PATH', default=''),
-                     environ.get('LD_LIBRARY_PATH', default='')]),
-           'IGN_GAZEBO_SYSTEM_PLUGIN_PATH':  # TODO(CH3): To support pre-garden. Deprecated.
-                      ':'.join([environ.get('IGN_GAZEBO_SYSTEM_PLUGIN_PATH', default=''),
-                                environ.get('LD_LIBRARY_PATH', default='')])}
+    model_path, plugin_path, media_path = GazeboRosPaths.get_paths()
+    env = {
+        "GZ_SIM_SYSTEM_PLUGIN_PATH": os.pathsep.join(
+            [
+                environ.get("GZ_SIM_SYSTEM_PLUGIN_PATH", default=""),
+                environ.get("LD_LIBRARY_PATH", default=""),
+                plugin_path,
+            ]
+        ),
+        "IGN_GAZEBO_SYSTEM_PLUGIN_PATH": os.pathsep.join(  # TODO(CH3): To support pre-garden. Deprecated.
+            [
+                environ.get("IGN_GAZEBO_SYSTEM_PLUGIN_PATH", default=""),
+                environ.get("LD_LIBRARY_PATH", default=""),
+                plugin_path,
+            ]
+        ),
+        "GZ_SIM_RESOURCE_PATH": os.pathsep.join(
+            [
+                environ.get("GZ_SIM_RESOURCE_PATH", default=""),
+                model_path,
```

Comment:
Are `model_path` and `media_path` being used exactly the same? If so we can simplify the code just by return one that merges both (resource_path) from `GazeboRosPaths`.

Reply:
I believe model and media paths are combined into resource path now, so I've simplified the code as you suggested.